src/vivarium_gates_mncnh/components/intervention.py

```diff
@@ -248,34 +243,65 @@ def on_initialize_simulants(self, pop: SimulantData) -> None:
 
         self.population_view.update(pop_update)
 
-    def on_time_step_cleanup(self, event: Event) -> None:
-        if self._sim_step_name() != SIMULATION_EVENT_NAMES.PREGNANCY:
-            return
-        pop = self.population_view.get(event.index)
-        attends_anc = pop[COLUMNS.ANC_ATTENDANCE] != ANC_ATTENDANCE_TYPES.NONE
-        anc_pop = pop.loc[attends_anc]
-
-        if (
-            INTERVENTION_SCENARIOS[self.scenario].ifa_mms_coverage
-            == models.ORAL_IRON_INTERVENTION.MMS
-        ):
-            pop_update = pd.DataFrame(
-                {COLUMNS.ORAL_IRON_INTERVENTION: models.ORAL_IRON_INTERVENTION.MMS},
-                index=anc_pop.index,
+    def on_time_step(self, event: Event) -> None:
+        if self._sim_step_name() == SIMULATION_EVENT_NAMES.FIRST_TRIMESTER_ANC:
```

Comment:
Nit: I think will be more readable if you refactor this into separate methods to be called on the two ANC time-steps.